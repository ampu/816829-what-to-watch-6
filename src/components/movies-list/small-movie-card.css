.small-movie-card__player {
  object-fit: fill;
}
