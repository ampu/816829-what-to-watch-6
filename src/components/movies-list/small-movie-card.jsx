import React from 'react';
import {Link, generatePath} from 'react-router-dom';
import PropTypes from 'prop-types';

import {MainPath} from '../../constants/paths';
import {useTimeoutActivator} from '../../hooks/use-timeout-activator';

import './small-movie-card.css';

const PLAY_TIMEOUT = 1000;

const SmallMovieCard = ({movie = {}}) => {
  const {
    title = ``,
    preview,
    videoPreview,
  } = movie;

  const [isActive, handleMouseEnter, handleMouseLeave] = useTimeoutActivator(PLAY_TIMEOUT);

  return (
    <article className="small-movie-card catalog__movies-card"
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}>

      <div className="small-movie-card__image">
        {!isActive && <img src={preview} alt={title} width="280" height="175"/>}
        {isActive && (
          <video className="small-movie-card__player"
            src={videoPreview} poster={preview}
            width="280" height="175"
            muted autoPlay/>
        )}
      </div>

      <h3 className="small-movie-card__title">
        <Link className="small-movie-card__link" to={generatePath(MainPath.MOVIE, movie)}>{title}</Link>
      </h3>
    </article>
  );
};

SmallMovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string,
  }),
  isActive: PropTypes.bool,
};

export default SmallMovieCard;
