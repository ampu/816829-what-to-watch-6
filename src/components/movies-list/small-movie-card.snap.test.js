import React from 'react';
import {Router} from 'react-router-dom';

import {render} from '@testing-library/react';

import {SmallMovieCard} from './small-movie-card';
import {createMemoryHistory} from 'history';

describe(`SmallMovieCard component`, () => {
  it(`Should render according to the snapshot`, () => {
    const movie = {
      id: `fakeId`,
      preview: `fakePreview`,
      videoPreview: `fakeVideoPreview`,
    };
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <SmallMovieCard movie={movie}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });
});
