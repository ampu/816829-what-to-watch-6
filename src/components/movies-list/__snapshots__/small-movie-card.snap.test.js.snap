// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard component Should render according to the snapshot 1`] = `
<div>
  <article
    class="small-movie-card catalog__movies-card"
  >
    <div
      class="small-movie-card__image"
    >
      <img
        alt=""
        height="175"
        src="fakePreview"
        width="280"
      />
    </div>
    <h3
      class="small-movie-card__title"
    >
      <a
        class="small-movie-card__link"
        href="/films/fakeId"
      />
    </h3>
  </article>
</div>
`;
